plugins/module_utils/network/sonic/config/bgp_af/bgp_af.py

```diff
@@ -1256,6 +1312,24 @@ def get_delete_commands_requests_for_replaced_overridden(self, want, have, state
                             afi_command['max_path'] = max_path_command
                             requests.extend(self.get_delete_max_path_requests(vrf_name, afi, safi, afi_command['max_path'], False, afi_command['max_path']))
 
+                    if afi_conf.get('import') and afi_conf['import'].get('vrf'):
+                        match_import = match_afi_cfg['import'] if 'import' in match_afi_cfg else {}
+                        if not match_import or (match_import and not match_import.get('vrf')):
+                            afi_command['import'] = afi_conf['import']
+                            requests.extend(self.get_delete_import_requests(vrf_name, afi, safi, afi_conf['import'], True, None))
+                        else:
```

Comment:
For "replaced" state handling, the user is allowed to replace just the VRF list (without affecting the route map configuration). The route map configuration, since it is a "simple" attribute (not a list or dict) should cause full replacement of the AF import_vrf config if it modified. (So the VRF list should be deleted if the route map value is modified.) If the route map configuration is not modified, however, the user is allowed to request replacement of only the VRF list and this should not affect an existing route map configuration for the affected AFI/SAFI.

The handling is different for overridden state, where any change in the BGP AF config should cause overriding of all current BGP_AF configuration (not just the BGP_AFs specified in the executing playbook, but all of them) with whatever is specified in the executing playbook.